features/networkmonitor/impl/src/main/kotlin/io/element/android/features/networkmonitor/impl/DefaultNetworkMonitor.kt

```diff
@@ -66,36 +92,70 @@ class DefaultNetworkMonitor @Inject constructor(
             }
         }
         trySendBlocking(connectivityManager.activeNetworkStatus())
-        val request = NetworkRequest.Builder()
-            .addCapability(NetworkCapabilities.NET_CAPABILITY_VALIDATED)
-            .build()
+        val request = NetworkRequest.Builder().build()
 
         connectivityManager.registerNetworkCallback(request, callback)
-        Timber.d("Subscribe")
+        Timber.tag(tag).d("Subscribe")
         awaitClose {
-            Timber.d("Unsubscribe")
+            Timber.tag(tag).d("Unsubscribe")
             connectivityManager.unregisterNetworkCallback(callback)
         }
     }
         .distinctUntilChanged()
         .debounce(300)
         .onEach {
-            Timber.d("NetworkStatus changed=$it")
+            Timber.tag(tag).d("NetworkStatus changed=$it")
         }
-        .stateIn(appCoroutineScope, SharingStarted.WhileSubscribed(), connectivityManager.activeNetworkStatus())
+        .shareIn(sessionCoroutineScope, SharingStarted.WhileSubscribed(), replay = 1)
 
     private fun ConnectivityManager.activeNetworkStatus(): NetworkStatus {
-        return activeNetwork?.let {
-            getNetworkCapabilities(it)?.getNetworkStatus()
-        } ?: NetworkStatus.Offline
+        return if (activeNetwork != null) NetworkStatus.Online else NetworkStatus.Offline
     }
 
-    private fun NetworkCapabilities.getNetworkStatus(): NetworkStatus {
-        val hasInternet = hasCapability(NetworkCapabilities.NET_CAPABILITY_VALIDATED)
-        return if (hasInternet) {
-            NetworkStatus.Online
+    @OptIn(ExperimentalCoroutinesApi::class)
+    override val connectivity: StateFlow<NetworkStatus> = networkStatus.flatMapLatest {
+        if (it == NetworkStatus.Online) {
+            checkHomeServerConnectivity()
         } else {
-            NetworkStatus.Offline
+            flowOf(NetworkStatus.Offline)
+        }
+    }.stateIn(sessionCoroutineScope, SharingStarted.WhileSubscribed(), NetworkStatus.Online)
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    private suspend fun checkHomeServerConnectivity(): Flow<NetworkStatus> = withContext(coroutineDispatchers.io) {
+        val homeServerUrl = sessionStore.getSession(sessionIdHolder.current.value)?.homeserverUrl?.removeSuffix("/")
```

Comment:
I think it would be safer to use the SDK, either to get the homeserver url, or (maybe better) to do the `head` request.